### PR TITLE
Added CHAIN_ID environment variable for connecting to blockchain endpoints

### LINKED ISSUE #2142: Verify EVM Chain ID
Context:
We have some EAs that reads on-chain data using the JSON-RPC API. We use Ethers for this. The RPC URL is defined with env var as `RPC_URL`.
Some EAs support multiple networks. In those cases, the RPC URL is defined as `[network]_RPC_URL`.

Task:
For all EAs that use Ethers to call a JSON-RPC API, add verification of the chain ID. With ethers, this is done by adding the chain ID in the initialization of the JSON-RPC provider, e.g.:

```typescript
new ethers.providers.JsonRpcProvider(rpcUrl)
// becomes...
new ethers.providers.JsonRpcProvider(rpcUrl, chainId)
```

To support this, we should add new env vars:
* `CHAIN_ID` for EAs that only support a single network
* `[network]_CHAIN_ID` for EAs that support multiple networks

The default for `CHAIN_ID` should be `1`, unless the EA is specifically intended for a different chain.

This would likely require a **major** version bump for the affected EAs, as if a node op is using the EA for a chain that isn't Ethereum mainnet, all requests would fail until re-configured with the correct chain ID.

### PR DESCRIPTION
**Pull request by karen-stepanyan**
## Quality Assurance

- [x] If a new adapter was made, or an existing one was modified so that its environment variables have changed, update the relevant `<ADAPTER_PACKAGE>/schemas/env.json` and `<ADAPTER_PACKAGE>/README.md`
- [ ] If a new adapter was made, or an existing one was modified so that its environment variables have changed, update the relevant `infra-k8s` configuration file.
- [x] The branch naming follows git flow (`feature/x`, `chore/x`, `release/x`, `hotfix/x`, `fix/x`) or is created from Clubhouse/Shortcut
- [x] This is related to a maximum of one Clubhouse/Shortcut story or GitHub issue
- [x] Types are safe (avoid TypeScript/TSLint features like any and disable, instead use more specific types)
- [x] All code changes have 100% unit and integration test coverage. If testing is not applicable or too difficult to justify doing, the reasoning should be documented explicitly in the PR.

### CODE REVIEW

File: .changeset/mean-jobs-hug.md
Comment (boxhock) [inline .changeset/mean-jobs-hug.md:35]:
```suggestion
Added CHAIN_ID environment variable and validation for connecting to blockchain endpoints. **WARNING:** Before upgrading, ensure the default CHAIN_ID value is correct for the chain(s) you use. If not, you need to explicitly set this env var. Please refer to the individual adapter README for more information.
```

File: packages/composites/anchor/README.md
Comment (boxhock) [inline packages/composites/anchor/README.md:12]: Just for consistency with how this is referred to in general. Needs to be updated in all places.

```suggestion
|           |        `ETHEREUM_CHAIN_ID`         |               The chain id to connect to               |                                                                                                 |                       1                        |
```

File: packages/sources/layer2-sequencer-health/src/config/index.ts
Comment (boxhock) [inline packages/sources/layer2-sequencer-health/src/config/index.ts:40]: Can we get these defaults set as constants? Makes it a bit easier to read and validate.

**Review by boxhock — CHANGES_REQUESTED** [review]
Great work! Have some comments

**Review by boxhock — COMMENTED** [review]
This looks good to me, but let's get a second pair of eyes on this PR as it contains breaking changes in a bunch of EAs.

File: packages/composites/dydx-rewards/README.md
Comment (KuphJr) [inline packages/composites/dydx-rewards/README.md:14]: Shouldn't the ETHEREUM_RPC_URL still be required here? (missing ✅)

File: packages/composites/dydx-rewards/schemas/env.json
Comment (KuphJr) [inline packages/composites/dydx-rewards/schemas/env.json:16]: Should this be required?

File: packages/composites/nftx/schemas/env.json
Comment (KuphJr) [inline packages/composites/nftx/schemas/env.json:13]: should this be required?

File: packages/sources/alpine/README.md
Comment (KuphJr) [inline packages/sources/alpine/README.md:9]: Should this be required?

File: packages/sources/alpine/src/endpoint/tvl.ts
Comment (KuphJr) [inline packages/sources/alpine/src/endpoint/tvl.ts:37]: Is it really safe to assume that if the chain isn't ETH, it is automatically Polygon?  Seems like we should also check if the network == POLYGON and throw an error if it is not ETH or POLYGON.

File: packages/sources/stader-labs/src/endpoint/maticx.ts
Comment (KuphJr) [inline packages/sources/stader-labs/src/endpoint/maticx.ts:28]: Why is the toString necessary?

File: packages/sources/stader-labs/src/endpoint/sftmx.ts
Comment (KuphJr) [inline packages/sources/stader-labs/src/endpoint/sftmx.ts:28]: why toString?

File: packages/sources/layer2-sequencer-health/src/config/index.ts
Comment (KuphJr) [inline packages/sources/layer2-sequencer-health/src/config/index.ts:41]: Not blocking approval, but can you explain this bracket syntax?

**Review by KuphJr — COMMENTED** [review]
Just a few questions before approval

File: packages/composites/dydx-rewards/README.md
Comment (karen-stepanyan) [reply packages/composites/dydx-rewards/README.md:14]: Oh, this one is mistake, thanks. Will fix

File: packages/composites/dydx-rewards/schemas/env.json
Comment (karen-stepanyan) [reply packages/composites/dydx-rewards/schemas/env.json:16]: Since there is a default value if not provided, we don't really need this variable to be required. Assuming most of the times connection is made to mainnet. Also for other adapters as well

File: packages/sources/alpine/src/endpoint/tvl.ts
Comment (karen-stepanyan) [reply packages/sources/alpine/src/endpoint/tvl.ts:37]: The ethers library itself will throw error if the chain is incorrect. Regarding the syntax i tried to use the same that is done for rpcUrl

File: packages/sources/stader-labs/src/endpoint/maticx.ts
Comment (karen-stepanyan) [reply packages/sources/stader-labs/src/endpoint/maticx.ts:28]: Haven't touch this part but i see that adapterSpecificParams have type  `[T: string]: string | number` so this is just for types

File: packages/sources/stader-labs/src/endpoint/sftmx.ts
Comment (karen-stepanyan) [reply packages/sources/stader-labs/src/endpoint/sftmx.ts:28]: Haven't touch this part but i see that adapterSpecificParams have type [T: string]: string | number so this is just for types

File: packages/sources/layer2-sequencer-health/src/config/index.ts
Comment (karen-stepanyan) [reply packages/sources/layer2-sequencer-health/src/config/index.ts:41]: I believe it's called Dynamic Property Key syntax. Networks is enum value (can be another object). Instead of `{'metis': '1088'}` we can use `{[Networks.Metis]: '1088'}` both are the same but second one in this particular case makes more sense because there are multiple places where `metis` is used as an object key .

File: packages/composites/dydx-rewards/schemas/env.json
Comment (boxhock) [reply packages/composites/dydx-rewards/schemas/env.json:16]: Let's make it not required as long as there's a default set.

File: packages/sources/layer2-sequencer-health/src/config/index.ts
Comment (KuphJr) [reply packages/sources/layer2-sequencer-health/src/config/index.ts:41]: Thanks for teaching me!

**Review by KuphJr — APPROVED** [review]

File: packages/sources/por-address-list/schemas/env.json
Comment (boxhock) [inline packages/sources/por-address-list/schemas/env.json:16]: Is this reflected in this EA's README?

Comment (boxhock) [reply packages/sources/por-address-list/schemas/env.json:16]: Does it get auto-generated after the PR is merged?

**Review by boxhock — APPROVED** [review]
I noticed that some READMEs weren't updated, but then I realized that they might be auto-generated retroactively?

If so, then this looks good to me! Great job on this :+1:

File: packages/sources/por-address-list/schemas/env.json
Comment (karen-stepanyan) [reply packages/sources/por-address-list/schemas/env.json:16]: Yes, i didn't change readmes where it gets generated automatically. According to [docs](https://github.com/smartcontractkit/external-adapters-js/tree/develop/packages/scripts#auto-generation-1) it should do it after merge to develop

Comment (austinborn) [reply packages/sources/por-address-list/schemas/env.json:16]: You should be able to see the changes once the changeset PR is updated after this is merged

**Pull request merged**
